common/src/main/java/com/microsoft/identity/common/internal/controllers/CommandDispatcher.java

```diff
@@ -439,6 +454,13 @@ private static void commandCallbackOnError(@SuppressWarnings("rawtypes") BaseCom
     @SuppressWarnings(WarningType.unchecked_warning)
     private static void commandCallbackOnTaskCompleted(@SuppressWarnings("rawtypes") BaseCommand command, CommandResult result) {
         command.getCallback().onTaskCompleted(result.getResult());
+        if (CommandResult.ResultStatus.REFRESH == result.getStatus()) {
+            if (command instanceof SilentTokenCommand) {
+                performRefresh(command);
+            } else {
+                throw new  IllegalArgumentException("Excepted type: SilentTokenCommand. Input was instead of type: " + command.toString());
+            }
+        }
```

Comment:
It seems like we are returning all the way back here to Command Dispatcher and based on the presence of specific flag called as `shouldRefresh` then we submit another command to the dispatcher for the sole purpose of refreshing the tokens....Is it possible to not come all the way back here to the Dispatcher and rather inside the controller when we are doing the token request, then if we see a need to refresh, can we spin up another thread there and just do the refresh there?

Reply:
But if it requires more complex changes then this is fine too...

Reply:
Outdated